components/script/dom/readablestream.rs

```diff
@@ -357,6 +357,49 @@ impl Callback for PipeTo {
 }
 
 impl PipeTo {
+    #[allow(clippy::too_many_arguments)]
+    fn delay_cancel_until_abort(
+        &self,
+        cx: SafeJSContext,
+        global: &GlobalScope,
+        source: &ReadableStream,
+        reason: SafeHandleValue,
+        abort_promise: Rc<Promise>,
+        realm: InRealm,
+        can_gc: CanGc,
+    ) -> Rc<Promise> {
+        // Spec demands the actions list run in order, so have cancel wait
+        // for abort promise before starting (to meet abort.any.js expectations).
+        let delayed_cancel = Promise::new(global, can_gc);
+        let triggered = Rc::new(Cell::new(false));
+        rooted!(in(*cx) let mut fulfillment_handler = Some(StartCancelAfterAbortHandler {
+            source: Dom::from_ref(source),
+            reason: RootedTraceableBox::new(Heap::default()),
+            result: delayed_cancel.clone(),
+            triggered: triggered.clone(),
+        }));
+        fulfillment_handler
```

Comment:
Instead of setting the reason afterwards, we can use `Heap::boxed(reason.get())`.